frontend/src/static/js/v_summary.js

```diff
@@ -145,21 +146,39 @@ window.vSummary = {
                 + `since=${slice.sinceDate}'T'00:00:00+08:00&`
                 + `until=${untilDate}'T'23:59:59+08:00`;
     },
-    getContributionBars(totalContribution) {
-      const res = [];
+    getContributionBars(fileTypeContribution) {
+      let totalWidth = 0;
       const contributionLimit = (this.avgContributionSize * 2);
+      const totalBars = {};
 
-      const cnt = parseInt(totalContribution / contributionLimit, 10);
-      for (let cntId = 0; cntId < cnt; cntId += 1) {
-        res.push(100);
-      }
+      Object.keys(fileTypeContribution).forEach((fileType) => {
+        const contribution = fileTypeContribution[fileType];
+        const res = [];
+        let fileTypeWidth = 0;
 
-      const last = (totalContribution % contributionLimit) / contributionLimit;
-      if (last !== 0) {
-        res.push(last * 100);
-      }
+        const cnt = parseInt(contribution / contributionLimit, 10);
+        for (let cntId = 0; cntId < cnt; cntId += 1) {
+          res.push(100);
```

Comment:
seems like 100 denote the max length of contribution bar. Perhaps make it a constant?